setuptools/tests/test_build_meta.py

```diff
@@ -132,7 +132,9 @@ def get_build_backend(self):
     @pytest.fixture(params=defns)
     def build_backend(self, tmpdir, request):
         path.build(request.param, prefix=str(tmpdir))
-        with tmpdir.as_cwd():
+        with tmpdir.as_cwd(), warnings.catch_warnings():
+            warnings.simplefilter("ignore", DeprecationWarning)
+            warnings.filterwarnings("ignore", "__legacy__ backend conflicts")
```

Comment:
Have you considered using a `with pytest.warns()` check instead of just suppressing them?

Reply:
I tried to remove all filter warnings I managed, and raise exceptions to avoid accidental behaviour.

Relevant commits: https://github.com/pypa/setuptools/pull/3064/commits/660a273f67dc0aa3e9d2818079051a4b927d91bf and https://github.com/pypa/setuptools/pull/3064/commits/f5fe86f2ac6162537699557067e1cbb2620b9226

This is related to https://github.com/pypa/setuptools/pull/3064#discussion_r803011890.

---

Rewriting this particular test in terms of `pytest.warns` is very complicated. When I was adding tests, I tried to keep the structure very similar to the existing code in the file to avoid being to disruptive and rewriting everything. Unfortunately the existing test uses some tricks, e.g. `TestBuildMetaLegacyBackend` inherits from `TestBuildMetaBackend` to reuse test cases, and fixtures are given via `pytest.fixture(params=...)` instead of `pytest.mark.parametrize`, etc...

This made it very difficult to not rely on warning filters on #3068... (although I tried, I did not manage to remove all the filters)